### PR TITLE
Background workers

### PR DESCRIPTION
**Pull request by Jbern16**
Created Whenever Cron Tab to Update Pending Transactions based on the pending_transactions pool
Updated Transactions to create when signed response is not 200
Created test to test update transaction functionality

### CODE REVIEW

File: app/models/transaction.rb [hunk 1]
```diff
@@ -1,16 +1,22 @@
+require 'transaction_update_service'
+
 class Transaction < ActiveRecord::Base
   validates_presence_of :from, :to, :amount
-  before_save :send_transaction
+  before_create :send_transaction
```
Comment (pindell-matt) [inline app/models/transaction.rb:5]: This is good, may want to check other models to see if anything else is still using before_save (or whatever else) when it'd be ideal to be more specific.

File: app/models/transaction.rb [hunk 2]
```diff
@@ -25,8 +31,31 @@ def time
...
+
+    if pending_transactions.empty?
+      completed = Transaction.pending
+    else
+      completed = pending_transactions.map do |pending|
+        Transaction.pending.where.not(from: pending[:from], to: pending[:to])
+      end.flatten
+    end
+
+    mark_complete(completed)
```
Comment (pindell-matt) [inline app/models/transaction.rb:46]: Nice refactor!

File: app/services/transaction_update_service.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  def current_pending
+    @connection.get do |req|
+      req.url '/pending_transactions'
+      req.headers['Content-Type'] = 'application/json'
+    end
+  end
+
+  def parse_current_pending
+    transactions = JSON.parse(current_pending.body)
```
Comment (pindell-matt) [inline app/services/transaction_update_service.rb]: Maybe put this inside the 'if Transaction.pending.any?' loop?

@pindell-matt  done.

File: app/models/transaction.rb [hunk 4]
```diff
@@ -1,16 +1,22 @@
...
   def send_transaction
     pkey     = find_pkey(self.from)
     service  = ClarkeService.new(pkey)
     unsigned = service.parsed_unsigned_payment(self.from, self.to, self.amount)
     signed   = service.parsed_signed_payment(unsigned)
+    status   = signed.status
+    rescue_if_failed unless status == 200
+  end
 
+  def rescue_if_failed
```
Comment (pindell-matt) [inline app/models/transaction.rb:18]: Nice.

**Comment by jdliss** [comment]
This is great, I'll add Twilio to this asap

**Pull request merged**